include/aspect/particle/property/interface.h

```diff
@@ -178,6 +178,19 @@ namespace aspect
           need_update () const;
 
           /**
+           * Return, which data has to be provided to update the property.
+           * Note that particle properties can only ask for no data, the
+           * solution values, and the solution gradients and all other update
+           * flags will throw exceptions.
+           *
+           * @return A vector with as many entries as solution components.
+           * Each entry contains the necessary update flags for this component.
```

Comment:
Oh, reading down further, I see now that it's not actually a vector with as many elements as there are *properties* here, but the *global number of field components*. I think that would be worth making clearer.

But that still raises the question whether you actually need this fine-grained a control over it.